extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php

```diff
@@ -190,6 +191,48 @@ function () {
 		return $botIds;
 	}
 
+
+	/**
+	 * @return array list of blacklisted ids for Top Contributors
+	 */
+	private function getBlacklistedIds(){
+		$blacklistedIds = WikiaDataAccess::cache(
+			self::getMemcKey( self::ALL_BLACKLISTED_IDS_MCACHE_KEY ),
+			WikiaResponse::CACHE_LONG,
+			function () {
+				global $wgExternalSharedDB;
+				$globalDb = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+				$globalIds = ( new WikiaSQL() )
+					->SELECT( 'ug_user' )
+					->FROM ( 'user_groups' )
+					->WHERE( 'ug_group' )->IN( [ 'bot', 'bot-global', 'staff', 'util', 'helper', 'vstf' ] )
+					->GROUP_BY( 'ug_user' )
+					->runLoop( $globalDb, function ( &$globalIds, $row ) {
+						$globalIds[] = $row->ug_user;
+					} );
+
+				$localDb = wfGetDB( DB_SLAVE );
+
+				$localUsers = ( new WikiaSQL() )
+					->SELECT( 'ug_user' )
+					->FROM ( 'user_groups' )
+					->WHERE( 'ug_group' )->NOT_IN( [ 'bot', 'bot-global' ] )
```

Comment:
Oh, bah, I missed that. ;)

So, you want to remove local users (who are also one of staff, helper, etc. globally) from being blacklisted if they have local rights (basically if they either have rollback or sysop rights)?

Also, you can remove `bot-global` from this query, that's only a global right, so shouldn't be in the local table.